### ISSUE TITLE
add BLF functionality (busy lamp field)

### ISSUE THREAD

**Issue by mokkin** [BelledonneCommunications/linphone-desktop#454]
A pretty nice feature on hardware phones is BLF (busy lamp field). You can program hardware keys to specific participants, see their call state immediately and even transfer calls by clicking on the key.

https://www.3cx.com/pbx/blf-function-keys/

Having this in Linphone would be great!

**Comment by InterLinked1** [comment]
The documentation says this should be supported: https://www.linphone.org/technical-corner/linphone?qt-technical_corner=1#qt-technical_corner
"Presence status"

Is this not the case?

**Comment by mokkin** [comment]
Actually I can't find it there. Or do you mean the entry "call transfer"?

**Comment by InterLinked1** [comment]
> Actually I can't find it there. Or do you mean the entry "call transfer"?

Never mind, I see what you mean. I think the presence panel can be used to change this status, but there's no way to actually see it.

**Comment by Didotto** [comment]
I'd like to know if the functionality of BLF (Busy Lamp Field) has been implemented. There's a specific indicator for that?

**Comment by olku** [comment]
> I'd like to know if the functionality of BLF (Busy Lamp Field) has been implemented. There's a specific indicator for that?

Yes, if you mean presence (as this is not a hardware device with physical LEDs)

**Comment by Didotto** [comment]
I mean the chance to see if a user is busy in the main interface. Imagine a use case where the secretary can see easly if his supervisor is busy and if not then she/he can choose to transfer a call.

**Comment by clst** [comment]
Linphone 2 has this feature and it works well with Asterisk/FreePBX.

**Comment by pete1019** [comment]
> Linphone 2 has this feature and it works well with Asterisk/FreePBX.

Can you please explain in more detail how to get BLF working with latest Linphone and Asterisk/FreePBX. Thanks!

**Comment by nopeee535** [comment]
are there any News ? 
we use Linphone 5.1.2 with placetel (cisco) and i found no way to activate the blf feature
Thanks

**Comment by Shaaah** [comment]
Does anyone have tips to configure BLF on Linphone with an asterisk/FreePBX server ?

**Comment by ComputerSnail** [comment]
Old versions of Linphone (e.g. from Debian package repository) supported BLF / presence status with Asterisk. Worked fine. 
But I can't get it to work with Linphone 5.x. 
As far as I know, this is due to Linphone 5.x dropping support for presence via SIP SIMPLE, and instead using a RFC 4662 Resource Lists server. See #84 here!
If I [set up a resource list in Asterisk as described in their documentation](https://docs.asterisk.org/Configuration/Channel-Drivers/SIP/Configuring-res_pjsip/Resource-List-Subscriptions-RLS/):
What will be the **RLS URI** I have to enter in the Linphone settings?
Did anybody here succeed in using Linphone 5.x with BLF and Asterisk?

**Comment by RussH** [comment]
Any chance that SIP SIMPLE will be re-added  as a supported  option?

**Comment by PHHENS** [comment]
Same need here : basic presence Subscribe/notify (event:dialog or presence).
Thanks
